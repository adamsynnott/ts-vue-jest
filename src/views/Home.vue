<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
    <hello-world v-if="toggle" msg="Welcome to Your Vue.js + TypeScript App"></hello-world>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import HelloWorld from '@/components/HelloWorld.vue'; // @ is an alias to /src

@Component({
  components: {
    'hello-world': HelloWorld,
  },
})
export default class Home extends Vue {
  @Prop({ default: false }) private toggle!: boolean;

  public togglePost() {
		this.toggle = !this.toggle;
  }
  
  get postToggle() {
    return this.toggle;
  }
}
</script>
