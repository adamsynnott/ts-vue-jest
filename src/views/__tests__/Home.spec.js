import { shallowMount } from '@vue/test-utils';
import Home from '../Home.vue';

describe('Home.vue', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallowMount(Home, {
      propsData: { 
        toggle: false,
       },
    });
  });
  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should change the state of the postToggle', () => {
    const context = { toggle: false };
    Home.options.methods.togglePost.call(context);
    expect(context.toggle).toBeTruthy();
  });
  it('should return the value of postToggle', () => {
    const context = { toggle: false };
    const postToggle = Home.options.computed.postToggle.get.call(context);
    expect(postToggle).toBeFalsy();
  });
});
