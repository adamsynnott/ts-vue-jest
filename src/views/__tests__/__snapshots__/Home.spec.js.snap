// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue matches snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="../assets/logo.png">
  <!---->
</div>
`;
